BitwardenShared/UI/Vault/VaultItem/ViewItem/ViewLoginItem/Extensions/CipherView+Update.swift

```diff
@@ -184,21 +194,55 @@ extension CipherView {
             viewPassword: viewPassword,
             localData: localData,
             attachments: attachments,
-            fields: addEditState.customFieldsState.customFields.isEmpty ?
-                nil : addEditState.customFieldsState.customFields.map { customField in
-                    FieldView(
-                        name: customField.name,
-                        value: customField.value,
-                        type: .init(fieldType: customField.type),
-                        linkedId: customField.linkedIdType?.rawValue
-                    )
-                },
+            fields: newFields,
             passwordHistory: passwordHistory,
             creationDate: creationDate,
             deletedDate: deletedDate,
             revisionDate: revisionDate
         )
     }
+
+    private func updatePasswordHistoryWithHiddenFields(
+        passwordHistory: [PasswordHistoryView]?,
+        oldFields: [FieldView]?,
+        newFields: [FieldView]?,
+        lastUsedDate: Date
+    ) -> [PasswordHistoryView]? {
+        guard let oldFields else {
+            return passwordHistory
+        }
+        let newPasswordHistoryFields: [PasswordHistoryView] = oldFields
+            .filter { field in
+                FieldType(fieldType: field.type) == FieldType.hidden &&
+                    !field.name.isEmptyOrNil &&
+                    !field.value.isEmptyOrNil &&
+                    !(newFields?.contains(field) ?? false)
+            }.compactMap { hiddenField in
+                guard !hiddenField.name.isEmptyOrNil else {
+                    return nil
+                }
+                return PasswordHistoryView(
+                    password: "\(hiddenField.name ?? ""): \(hiddenField.value ?? "")",
+                    lastUsedDate: lastUsedDate
+                )
+            }
```

Comment:
🤔 Sorry I missed this before but given that in the `filter` you have `&& !field.name.isEmptyOrNil`; is it necessary to check again and add the guard in `compactMap`?
```swift
guard !hiddenField.name.isEmptyOrNil else {
      return nil
}
```

🤔 And another point I missed: if you have a Field that now has changed its **name** then `newFields.contains(field)` will be `false` and will be included in the history, adding the `PasswordHistoryView` but using the old field name instead of the new one. This is not the same behavior as in MAUI where we only added the password history if the names of the old field and the new field would match. Maybe something for Product to make decision on.

https://github.com/bitwarden/mobile/blob/44849cb0a25019f240bc0064048a90f063000a71/src/Core/Services/CipherService.cs#L160-L169

Reply:
Ran it with product and we will keep consistency with the other clients. So if the name changes it will log old name + old password.

Reply:
https://github.com/user-attachments/assets/025c7f15-c5ab-42f9-acd0-655d41ef2e73

Reply:
Removed the nil name check because it was not necessary